src/Bridges/bridge_optimizer.jl

```diff
@@ -1143,8 +1147,11 @@ function MOI.set(
     b::AbstractBridgeOptimizer,
     attr::MOI.ConstraintSet,
     ci::MOI.ConstraintIndex{MOI.VariableIndex,S},
-    value::S,
+    value,
 ) where {S<:MOI.AbstractScalarSet}
+    if typeof(value) != S
+        throw(ArgumentError("Invalid type when setting ConstraintSet."))
```

Comment:
Is the plan to add this in every model implementing MOI? Why not fixing it for all of them at the same time in `src/attributes.jl`

Reply:
Because generic fallbacks get in the way.